docker-compose.yml

```diff
@@ -1,11 +1,16 @@
 version: '3'
 services:
   boulder:
-    image: &boulder_image letsencrypt/boulder-tools:${BOULDER_TOOLS_TAG:-go1.17.7_2022-02-10}
+    image: &boulder_image letsencrypt/boulder-tools:${BOULDER_TOOLS_TAG:-go1.18beta2_2022-03-08}
     environment:
       FAKE_DNS: 10.77.77.77
       BOULDER_CONFIG_DIR: test/config
       GOFLAGS: -mod=vendor
+      # Go 1.18 turns off SHA-1 validation on CSRs (and certs, but that doesn't
+      # affect us). It also turns off TLS 1.0 and TLS 1.1. Temporarily go back
+      # to allowing these so we can upgrade to Go 1.18 while doing a deprecation
+      # window. These overrides will stop working in Go 1.19.
+      GODEBUG: x509sha1=1,tls10default=1
```

Comment:
My only question is: how do we ensure that SRE uses this same env var to build Production boulder? (And, presumably, *not* use it to build Staging boulder?)

Reply:
Good question. In the past we've just filed a separate ticket to add it to the production env. But that's a little fragile / obscure. I'll see if we can set GODEBUG from within the program and still have it take effect. That would allow us to use our feature flag infra instead.

Reply:
Might be tough. On the one hand, it [just uses `os.GetEnv`](https://github.com/golang/go/blob/master/src/internal/godebug/godebug.go) at runtime to determine the value of GODEBUG, but crypto/tls reads that value [when first loaded](https://github.com/golang/go/blob/2580d0e08d5e9f979b943758d3c49877fb2324cb/src/crypto/tls/common.go#L978) which may be before our features scaffolding can kick in.

Reply:
Yeah, I would not like to rely on init ordering for this to work correctly. Let's go ahead with setting this via env var in prod. I'll file a ticket.